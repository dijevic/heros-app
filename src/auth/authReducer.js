import { types } from "../tipos/types";

export const authReducer = (state = {}, action) => {

    switch (action.type) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }
        case types.logOut:
            return { logged: false }

        default:
            return state
    }
}