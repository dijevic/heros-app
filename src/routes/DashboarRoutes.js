import React from 'react'
import {
    Switch,
    Route,
    Redirect

} from "react-router-dom";
import { DcScreen } from '../components/dc/DcScreen'
import { HeroeScreen } from '../components/heroe/HeroeScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { SearchScreen } from '../components/search/SearchScreen';
import { Navbar } from '../components/ui/NavBar';

export const DashboarRoutes = () => {
    return (
        <>

            <Navbar />
            <div className="container mt-3">
                <Switch>
                    <Route exact component={MarvelScreen} path='/marvel' />
                    <Route exact component={HeroeScreen} path='/hero/:id' />
                    <Route exact component={DcScreen} path='/dc' />
                    <Route exact component={SearchScreen} path='/search' />

                    <Redirect to='/marvel' />
                </Switch>
            </div>
        </>
    )
}
