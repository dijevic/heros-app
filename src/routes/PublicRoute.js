import React from 'react'
import PropTypes from 'prop-types'

import { Route, Redirect } from 'react-router-dom'


export const PublicRoute = ({ component: Component, isAuth, ...rest }) => {


    return (
        <Route {...rest} component={(props) => (
            (!isAuth)
                ? <Component {...props} />
                : <Redirect to="/" />
        )
        } />
    )
}

PublicRoute.propTypes = {
    component: PropTypes.func.isRequired,
    isAuth: PropTypes.bool.isRequired
}