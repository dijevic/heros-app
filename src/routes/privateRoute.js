import React from 'react'
import PropTypes from 'prop-types'

import { Route, Redirect } from 'react-router-dom'


export const PrivateRoute = ({ component: Component, isAuth, ...rest }) => {

    localStorage.setItem('lastPath', rest.location.pathname)
    return (
        <Route {...rest} component={(props) => (
            (isAuth)
                ? <Component {...props} />
                : <Redirect to="/login" />
        )
        } />
    )
}

PrivateRoute.propTypes = {
    component: PropTypes.func.isRequired,
    isAuth: PropTypes.bool.isRequired
}
