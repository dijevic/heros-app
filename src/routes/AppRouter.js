import React, { useContext } from "react";
import {
    BrowserRouter as Router,
    Switch
} from "react-router-dom";
import { authContext } from "../auth/AuthContext";
import { LoginScreen } from "../components/login/LoginScreen";

import { DashboarRoutes } from "./DashboarRoutes";
import { PrivateRoute } from "./privateRoute";
import { PublicRoute } from "./PublicRoute";


export const AppRouter = () => {
    const { usuario } = useContext(authContext)
    return (
        <Router>
            <div >

                {/* llamo a mi navbar que ya tinene los links implementados */}
                {/* <Navbar /> */}
                <Switch>
                    <PublicRoute component={LoginScreen} path="/login" exact isAuth={usuario.logged} />
                    <PrivateRoute component={DashboarRoutes} path="/" isAuth={usuario.logged} />
                </Switch>
            </div>
        </Router>
    )
}
