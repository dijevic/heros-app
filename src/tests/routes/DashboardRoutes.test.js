import { mount } from "enzyme"
import { MemoryRouter } from "react-router"
import { authContext } from "../../auth/AuthContext"
import { DashboarRoutes } from "../../routes/DashboarRoutes"

describe('pruebas en mi componente dashboard routes', () => {
    const context = {
        usuario: {
            logged: true,
            nombre: 'diego vielma'
        },
        dispatch: jest.fn()
    }

    test('debe de hacer match con el snapshot', () => {
        const wrapper = mount(

            <authContext.Provider value={context}>
                <MemoryRouter>
                    <DashboarRoutes />

                </MemoryRouter>
            </authContext.Provider>)

        expect(wrapper).toMatchSnapshot()
    })
})