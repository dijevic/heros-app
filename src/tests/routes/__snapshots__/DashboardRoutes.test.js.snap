// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en mi componente dashboard routes debe de hacer match con el snapshot 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark "
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Heroes
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 "
    >
      <ul
        className="navbar-nav ml-auto d-flex "
      >
        <p
          className="nav-item nav-link name"
        >
          <b>
            diego vielma
          </b>
        </p>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-3"
  >
    <div>
      <h1>
        marvel Comics || Heroes
      </h1>
      <hr
        className="mt-3"
      />
      <div
        className="card-columns row row-cols-1 row-cols-md-3 g-4"
      >
        <div
          className="my-card"
          onClick={[Function]}
        >
          <img
            alt="Spider Man"
            className="img img-responsive"
            src="./assets/heroes/marvel-spider.jpg"
          />
          <div
            className="profile-name"
          >
            Spider Man
          </div>
          <div
            className="profile-position"
          >
            Peter Parker
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Amazing Fantasy #15
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="my-card"
          onClick={[Function]}
        >
          <img
            alt="Captain America"
            className="img img-responsive"
            src="./assets/heroes/marvel-captain.jpg"
          />
          <div
            className="profile-name"
          >
            Captain America
          </div>
          <div
            className="profile-position"
          >
            Steve Rogers
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Captain America Comics #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="my-card"
          onClick={[Function]}
        >
          <img
            alt="Iron Man"
            className="img img-responsive"
            src="./assets/heroes/marvel-iron.jpg"
          />
          <div
            className="profile-name"
          >
            Iron Man
          </div>
          <div
            className="profile-position"
          >
            Tony Stark
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Tales of Suspense #39
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="my-card"
          onClick={[Function]}
        >
          <img
            alt="Thor"
            className="img img-responsive"
            src="./assets/heroes/marvel-thor.jpg"
          />
          <div
            className="profile-name"
          >
            Thor
          </div>
          <div
            className="profile-position"
          >
            Thor Odinson
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Journey into Myster #83
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="my-card"
          onClick={[Function]}
        >
          <img
            alt="Hulk"
            className="img img-responsive"
            src="./assets/heroes/marvel-hulk.jpg"
          />
          <div
            className="profile-name"
          >
            Hulk
          </div>
          <div
            className="profile-position"
          >
            Bruce Banner
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    The Incredible Hulk #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="my-card"
          onClick={[Function]}
        >
          <img
            alt="Wolverine"
            className="img img-responsive"
            src="./assets/heroes/marvel-wolverine.jpg"
          />
          <div
            className="profile-name"
          >
            Wolverine
          </div>
          <div
            className="profile-position"
          >
            James Howlett
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    The Incredible Hulk #180
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="my-card"
          onClick={[Function]}
        >
          <img
            alt="Daredevil"
            className="img img-responsive"
            src="./assets/heroes/marvel-daredevil.jpg"
          />
          <div
            className="profile-name"
          >
            Daredevil
          </div>
          <div
            className="profile-position"
          >
            Matthew Michael Murdock
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Daredevil #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="my-card"
          onClick={[Function]}
        >
          <img
            alt="Hawkeye"
            className="img img-responsive"
            src="./assets/heroes/marvel-hawkeye.jpg"
          />
          <div
            className="profile-name"
          >
            Hawkeye
          </div>
          <div
            className="profile-position"
          >
            Clinton Francis Barton
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Tales of Suspense #57
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="my-card"
          onClick={[Function]}
        >
          <img
            alt="Cyclops"
            className="img img-responsive"
            src="./assets/heroes/marvel-cyclops.jpg"
          />
          <div
            className="profile-name"
          >
            Cyclops
          </div>
          <div
            className="profile-position"
          >
            Scott Summers
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    X-Men #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="my-card"
          onClick={[Function]}
        >
          <img
            alt="Silver Surfer"
            className="img img-responsive"
            src="./assets/heroes/marvel-silver.jpg"
          />
          <div
            className="profile-name"
          >
            Silver Surfer
          </div>
          <div
            className="profile-position"
          >
            Norrin Radd
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    The Fantastic Four #48
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
